src/main/java/org/mitre/synthea/export/BB2RIFExporter.java

```diff
@@ -432,6 +447,66 @@ public void export(Person person, long stopTime, int yearsOfHistory) throws IOEx
     rifWriters.getOrCreateWriter(EXPORT_SUMMARY.class, -1, "csv", ",").writeValues(exportCounts);
   }
 
+  private enum ClaimType {
+    CARRIER,
+    OUTPATIENT,
+    INPATIENT,
+    HHA,
+    DME,
+    SNF,
+    PDE,
+    HOSPICE
+  }
+
+  private boolean isVAorIHS(HealthRecord.Encounter encounter) {
+    boolean isVA = (ProviderType.VETERAN == encounter.provider.type);
+    // IHS facilities have valid 6 digit id, IHS centers don't
+    boolean isIHSCenter = (ProviderType.IHS == encounter.provider.type)
+            && encounter.provider.id.length() != 6;
+    return isVA || isIHSCenter;
+  }
+
+  private Set<ClaimType> getClaimTypes(HealthRecord.Encounter encounter) {
+    Set<ClaimType> types = new HashSet<>();
+
+    boolean isSNF = encounter.type.equals(EncounterType.SNF.toString());
+    boolean isHome = encounter.type.equals(EncounterType.HOME.toString());
+    boolean isHospice = encounter.type.equals(EncounterType.HOSPICE.toString());
+    boolean isInpatient = encounter.type.equals(EncounterType.INPATIENT.toString());
+    boolean isEmergency = encounter.type.equals(EncounterType.EMERGENCY.toString());
+    boolean isWellness = encounter.type.equals(EncounterType.WELLNESS.toString());
+    boolean isUrgent = encounter.type.equals(EncounterType.URGENTCARE.toString());
+    boolean isVirtual = encounter.type.equals(EncounterType.VIRTUAL.toString());
+    boolean isAmbulatory = encounter.type.equals(EncounterType.AMBULATORY.toString());
+    boolean isOutpatient = encounter.type.equals(EncounterType.OUTPATIENT.toString());
+
+    boolean isPrimary = (ProviderType.PRIMARY == encounter.provider.type);
+    boolean isVirtualOutpatient = isVirtual
+            && (ProviderType.HOSPITAL == encounter.provider.type);
+
+    if (!isVAorIHS(encounter)) {
+      if (isSNF) {
+        types.add(ClaimType.SNF);
+      } else if (isHome) {
+        types.add(ClaimType.HHA);
+      } else if (isHospice) {
+        types.add(ClaimType.HOSPICE);
+      } else if (isInpatient || isEmergency) {
+        types.add(ClaimType.INPATIENT);
+      } else if (isPrimary || isWellness || isUrgent) {
+        types.add(ClaimType.CARRIER);
+      } else if (isAmbulatory || isOutpatient || isVirtualOutpatient) {
+        types.add(ClaimType.OUTPATIENT);
+      }
+
+      if (types.isEmpty()) {
+        System.out.printf("BFD RIF unhandled encounter type");
```

Comment:
```suggestion
        System.out.printf("BFD RIF unhandled encounter type (" + encounter.type
            + ") and provider type (" + encounter.provider.type.toString() + ")");
```

Just in case this happens... say if we add new encounter types or provider types...  it would be nice to print out the odd combination.